app/upgrades/v9/upgrades.go

```diff
@@ -13,8 +17,24 @@ func CreateUpgradeHandler(
 	configurator module.Configurator,
 	keepers *keepers.AppKeepers,
 ) upgradetypes.UpgradeHandler {
-	return func(ctx sdk.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+	return func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
 		ExecuteProp214(ctx, keepers.GAMMKeeper)
-		return mm.RunMigrations(ctx, configurator, vm)
+
+		// Add Interchain Accounts host module
+		// set the ICS27 consensus version so InitGenesis is not run
+		fromVM[icatypes.ModuleName] = mm.Modules[icatypes.ModuleName].ConsensusVersion()
+
+		// create ICS27 Controller submodule params, controller module not enabled.
+		controllerParams := icacontrollertypes.Params{}
+
+		// create ICS27 Host submodule params
+		hostParams := icahosttypes.Params{
+			HostEnabled:   true,
+			AllowMessages: []string{"/cosmos.bank.v1beta1.MsgSend"},
```

Comment:
To be safe here, what we can/should do, is the use the types directly.

i.e.

```go
AllowMessages: []string{
  sdk.MsgTypeURL(&banktypes.MsgSend{}),
  // ...
}
```

Reply:
such sad that we have to do this :/

Reply:
I'm going to add send, gov vote, stake, reward claim, and gamm messages.

I think we just go change the IBC interface to make it more sane to include everything by default >:( (Aditya mentioned this is on their roadmap, may be we can just help with a PR here, if they'd be down to release sooner)

Reply:
And Authz, they can take the L of executing two messages via authz'ing to themself lol